pkg/manifests/manifests_test.go

```diff
@@ -1188,6 +1164,66 @@ func TestPrometheusK8sRemoteWriteURLs(t *testing.T) {
 	}
 }
 
+func TestRemoteWriteAuthorizationConfig(t *testing.T) {
+	c, err := NewConfigFromString(`prometheusK8s:
```

Comment:
(nit) rather than a giant configuration testing all possibilities, I would rather see table-driven test cases.
For flexible verification, you can pass a custom function that would examine the resulting `Prometheus` object

```
for _, tc := range []struct {
    conf string
    checkFn func(*testing.T, []monv1.Prometheus)
} {
...
} {
   ...
   t.Run("", function(t *testing.T) {
      ...
      tc.checkFn(t, p)
   })
}
```

Reply:
Indeed, this way it's better, please do let me know if now it's ok